test/workbook/worksheet/tc_worksheet.rb

```diff
@@ -475,6 +464,16 @@ def test_name_size
     assert_nothing_raised { @ws.name = Array.new(31, "A").join() }
   end
 
+  def test_exception_if_name_too_long_because_of_multibyte_characters
+    four_characters_for_excel = "\u{1F1EB 1F1F7}" # french flag emoji
+    assert_raises(ArgumentError, "name too long!") do
+      @ws.name = "#{four_characters_for_excel}1234567890123456789012345678"
```

Comment:
Might be a bit easier to grasp if written like this:

```suggestion
      @ws.name = four_characters_for_excel + "x" * 28
```